javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -110,15 +111,19 @@ private Array(Object[] delegate) {
      *
      * @param <T>      Component type of the Array.
      * @param elements An Iterable of elements.
-     * @return A Array containing the given elements in the same order.
+     * @return An Array containing the given elements in the same order.
      * @throws NullPointerException if {@code elements} is null
      */
     @SuppressWarnings("unchecked")
     public static <T> Array<T> ofAll(Iterable<? extends T> elements) {
         Objects.requireNonNull(elements, "elements is null");
-        return elements instanceof Array
-               ? (Array<T>) elements
-               : wrap(toArray(elements));
+        if (elements instanceof Array) {
+            return (Array<T>) elements;
+        } else if (elements instanceof SeqAsJavaList) {
```

Comment:
we should provide a constant time converter, back to the `Seq` it was created from :)